asf_heat_pump_suitability/analysis/spatial_signatures_hn_zones/20251205_plymouth_investigation.py

```diff
@@ -0,0 +1,530 @@
+    )
+)
+plymouth_uprns_spatial_signature_gdf = uprns.generate_gdf_uprn_coords(
+    plymouth_uprns_spatial_signature_df
+)
+
+# %%
+# Filter spatial signatures in Plymouth to selected types
+selected_spatial_signatures_plymouth_gdf = spatial_signatures_plymouth_gdf[
+    spatial_signatures_plymouth_gdf["type"].isin(city_centre_types)
+]
+
+# %%
+# Inspect selected spatial signatures in relation to existing HNZ
+
+# colour for spatial signature types
+unique_types = selected_spatial_signatures_plymouth_gdf["type"].unique()
+colors = plt.cm.tab20.colors
+color_dict = {t: colors[i % len(colors)] for i, t in enumerate(unique_types)}
+
+
+fig, ax = plt.subplots(figsize=(15, 7.5))
+
+# plot spatial signatures polygons with mapped colors
+selected_spatial_signatures_plymouth_gdf.plot(
+    ax=ax, color=selected_spatial_signatures_plymouth_gdf["type"].map(color_dict)
+)
+
+# overlay Plymouth LA boundaries
+plymouth_la_boundaries_gdf.plot(ax=ax, facecolor="none", edgecolor="black", linewidth=1)
+
+# overlay Plymouth heat network zones
+plymouth_hn_zones_gdf.plot(ax=ax, facecolor="none", edgecolor="blue", linewidth=1)
+
+# manual legend for spatial signature types
+legend_handles = [mpatches.Patch(color=color_dict[t], label=t) for t in unique_types]
+
+# lines for boundaries and heat network zones
+legend_handles += [
+    Line2D([0], [0], color="black", lw=2, label="Plymouth LA boundary"),
+    Line2D([0], [0], color="blue", lw=2, label="Existing heat network zones"),
+]
+
+ax.legend(
+    handles=legend_handles,
+    title="Spatial Signature Type",
+    loc="upper left",
+    bbox_to_anchor=(1.05, 1),
+)
+
+plt.show()
+
+# %% [markdown]
+# 05/12/25: Only 3 of the 6 city centre spatial signature types are present within Plymouth LA boundaries (Dense urban neighbourhoods, Local urbanity and Regional urbanity). Only the existing HNZ PLYM-011 contain any of these city centre types. The remaining 7 existing HN zones are in a wide range of other spatial signature types.
+
+# %% [markdown]
+# **Comparison of zones by spatial coverage**
+
+# %%
+# Comparing total spatial overlap
+overlap = gpd.overlay(
+    plymouth_hn_zones_gdf, selected_spatial_signatures_plymouth_gdf, how="intersection"
+)
+overlap_area = overlap.geometry.area.sum()
+
+# % overlap of each layer
+area_hnz = plymouth_hn_zones_gdf.geometry.area.sum()
+area_selected = selected_spatial_signatures_plymouth_gdf.geometry.area.sum()
+
+percent_hnz_covered_by_selected = (overlap_area / area_hnz) * 100
+percent_selected_covered_by_hnz = (overlap_area / area_selected) * 100
+
+# summary table
+summary_df = pd.DataFrame(
+    {
+        "Area (m²)": [area_hnz, area_selected, overlap_area],
+        "% of HNZ covered by selected signatures": [
+            "100%",
+            f"{percent_selected_covered_by_hnz:.1f}%",
+            f"{(overlap_area / area_hnz) * 100:.1f}%",
+        ],
+        "% of selected signatures covered by HNZ": [
+            f"{(area_hnz / area_selected) * 100:.1f}%",
```

Comment:
This is just a size comparison right because there's no overlay involved and doesn't actually calculate percentage coverage?

Reply:
Ah good spot! This is a mistake. I've now corrected it to:
```suggestion
            f"{percent_hnz_covered_by_selected:.1f}%",
```

I've also cleaned up the summary table so it's clearer.